### PR TITLE
Added BOM UOM selection for items

### PR DESCRIPTION
**Pull request by bcornwellmott**
Adding fields for UOM and qty, and renaming the current qty field to stock_qty (in BOM as well as all related files). 

Based on feedback from:  https://github.com/frappe/erpnext/pull/7923

### CODE REVIEW

**Comment by bcornwellmott** [comment]
Here is a screenshot of the updated BOM Item table view. I've moved the row quantity and the stock qty related items to separate columns to make clear which ones belong to which. 
image

**Comment by bcornwellmott** [comment]
I'm not sure why the Travis failed here... any assistance would be helpful.

**Comment by bcornwellmott** [comment]
Anything holding this up?

**Comment by bcornwellmott** [comment]
I'm not sure why I'm failing the test cases, because I've updated them to work. 
I'm also not sure why cur_frm is failing, unless we are expected to update any code to eliminate cur_frm that's already there. My patch didn't add any cur_frm references. This all needs to be one patch in order to not break the current BOM functionality.

I've added a gif here:
ew bom 120170615092508

**Comment by rmehta** [comment]
@bcornwellmott sorry that was just a script I was testing. We will test and merge this PR, but from next PR we will be more strict with the Contribution Guidelines! As of now the cost of a small bug introduced inadvertently is more that the contribution, as users are concerned about stability. 

We are hoping once we start enforcing better testing both manual and automated, we hope the quality of releases will be much better.

Thanks for your contributions 👍. Keep them coming 😄

File: erpnext/patches/v8_0/update_stock_qty_value_in_bom_item.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# License: GNU General Public License v3. See license.txt
+
+from __future__ import unicode_literals
+import frappe
+
+def execute():
+	frappe.reload_doc('manufacturing', 'doctype', 'bom_item')
+	frappe.reload_doc('manufacturing', 'doctype', 'bom_explosion_item')
+	frappe.reload_doc('manufacturing', 'doctype', 'bom_scrap_item')
+	frappe.db.sql("update `tabBOM Item` set stock_qty = qty, uom = stock_uom")
```
Comment (mbauskar) [inline erpnext/patches/v8_0/update_stock_qty_value_in_bom_item.py]: @bcornwellmott ,

conversion_factor is set to 0 after migrate also, update `conversion_factor`

**Review by mbauskar — DISMISSED** [review]
Also can you please check

bom_item

**Comment by bcornwellmott** [comment]
I'm not able to replicate that issue, and it doesn't appear to be related to the changes I made. It looks like it can't seem to find the "conversion_rate" in validate_conversion_rate (from get_item_details.py). Given that I didn't change that code at all, I'm not sure why it would have any bugs. It looks like the validate_conversion_rate has a bug where it checks to see if conversion_rate is in args, but then tries to grab the precision from the conversion_rate in the document (which may not be there). 

I don't see how fixing that is related to this issue.

**Comment by bcornwellmott** [comment]
Is there anything I can do to help get this merged?

**Comment by rmehta** [comment]
@bcornwellmott sorry this week for sure. @nabinhait was on extended leave, so we were waiting for him. I will add the milestone.

**Review by rohitwaghchaure — APPROVED** [review]

**Pull request merged**